### ISSUE TITLE
System.Net.Sockets.SocketException: Address already in use

### ISSUE THREAD

**Issue by adamchester** [ionide/ionide-fsharp#173]
I get the following error in the dev tools when attempting to open two separate atom instances, both using F#, at the same time:

```
Unhandled Exception:
System.Net.Sockets.SocketException: Address already in use
   at System.Net.Sockets.Socket.Bind (System.Net.EndPoint local_end) in <filename unknown>:line 0
   at Suave.Tcp+startTcpIpServerAsync@118.Invoke (Microsoft.FSharp.Core.Unit unitVar0) in <filename unknown>:line 0
   at Suave.Tcp.run@94[a] (Microsoft.FSharp.Core.FSharpFunc`2 f, UInt16 _arg1) in <filename unknown>:line 0
   at Suave.Tcp.run@94[a] (Microsoft.FSharp.Core.FSharpFunc`2 f, UInt16 _arg1) in <filename unknown>:line 0
   at Suave.Tcp.run@94[a] (Microsoft.FSharp.Core.FSharpFunc`2 f, UInt16 _arg1) in <filename unknown>:line 0
   at Suave.Tcp.aFewTimes[a] (Microsoft.FSharp.Core.FSharpFunc`2 f) in <filename unknown>:line 0
   at Suave.Tcp.startTcpIpServerAsync (Int32 bufferSize, Int32 maxConcurrentOps, Logger logger, Microsoft.FSharp.Core.FSharpFunc`2 serveClient, Suave.Sockets.SocketBinding binding) in <filename unknown>:line 0
   at Suave.Web+ParsingAndControl.startWebWorkerAsync (Int32 bufferSize, Int32 maxOps, Microsoft.FSharp.Core.FSharpFunc`2 webpart, Suave.HttpRuntime runtime) in <filename unknown>:line 0
   at Suave.Web+servers@658.Invoke (Suave.HttpRuntime runtime) in <filename unknown>:line 0
   at Suave.Web+servers@657-1.Invoke (Suave.HttpBinding x) in <filename unknown>:line 0
   at Microsoft.FSharp.Primitives.Basics.List.map[T,TResult] (Microsoft.FSharp.Core.FSharpFunc`2 mapping, Microsoft.FSharp.Collections.FSharpList`1 x) in <filename unknown>:line 0
   at Microsoft.FSharp.Collections.ListModule.Map[T,TResult] (Microsoft.FSharp.Core.FSharpFunc`2 mapping, Microsoft.FSharp.Collections.FSharpList`1 list) in <filename unknown>:line 0
   at Suave.Web.startWebServerAsync (Suave.SuaveConfig config, Microsoft.FSharp.Core.FSharpFunc`2 webpart) in <filename unknown>:line 0
   at Suave.Web.startWebServer (Suave.SuaveConfig config, Microsoft.FSharp.Core.FSharpFunc`2 webpart) in <filename unknown>:line 0
   at FsAutoComplete.Suave.main (System.String[] argv) in <filename unknown>:line 0
[ERROR] FATAL UNHANDLED EXCEPTION: System.Net.Sockets.SocketException: Address already in use
   at System.Net.Sockets.Socket.Bind (System.Net.EndPoint local_end) in <filename unknown>:line 0
   at Suave.Tcp+startTcpIpServerAsync@118.Invoke (Microsoft.FSharp.Core.Unit unitVar0) in <filename unknown>:line 0
   at Suave.Tcp.run@94[a] (Microsoft.FSharp.Core.FSharpFunc`2 f, UInt16 _arg1) in <filename unknown>:line 0
   at Suave.Tcp.run@94[a] (Microsoft.FSharp.Core.FSharpFunc`2 f, UInt16 _arg1) in <filename unknown>:line 0
   at Suave.Tcp.run@94[a] (Microsoft.FSharp.Core.FSharpFunc`2 f, UInt16 _arg1) in <filename unknown>:line 0
   at Suave.Tcp.aFewTimes[a] (Microsoft.FSharp.Core.FSharpFunc`2 f) in <filename unknown>:line 0
   at Suave.Tcp.startTcpIpServerAsync (Int32 bufferSize, Int32 maxConcurrentOps, Logger logger, Microsoft.FSharp.Core.FSharpFunc`2 serveClient, Suave.Sockets.SocketBinding binding) in <filename unknown>:line 0
   at Suave.Web+ParsingAndControl.startWebWorkerAsync (Int32 bufferSize, Int32 maxOps, Microsoft.FSharp.Core.FSharpFunc`2 webpart, Suave.HttpRuntime runtime) in <filename unknown>:line 0
   at Suave.Web+servers@658.Invoke (Suave.HttpRuntime runtime) in <filename unknown>:line 0
   at Suave.Web+servers@657-1.Invoke (Suave.HttpBinding x) in <filename unknown>:line 0
   at Microsoft.FSharp.Primitives.Basics.List.map[T,TResult] (Microsoft.FSharp.Core.FSharpFunc`2 mapping, Microsoft.FSharp.Collections.FSharpList`1 x) in <filename unknown>:line 0
   at Microsoft.FSharp.Collections.ListModule.Map[T,TResult] (Microsoft.FSharp.Core.FSharpFunc`2 mapping, Microsoft.FSharp.Collections.FSharpList`1 list) in <filename unknown>:line 0
   at Suave.Web.startWebServerAsync (Suave.SuaveConfig config, Microsoft.FSharp.Core.FSharpFunc`2 webpart) in <filename unknown>:line 0
   at Suave.Web.startWebServer (Suave.SuaveConfig config, Microsoft.FSharp.Core.FSharpFunc`2 webpart) in <filename unknown>:line 0
   at FsAutoComplete.Suave.main (System.String[] argv) in <filename unknown>:line 0
(anonymous function) @ /Users/adamchester/.atom/packages/ionide-fsharp/lib/fsharp.js:2693
```

**Comment by nelak** [comment]
Is there a way to switch the port number?
Having anything else running at port: 8088 seems to break it also.

**Comment by haf** [comment]
Yes, the easiest way is probably to start the server with an incrementing port number until it starts properly. And there's a property on SocketException called something like SocketError which contains an enumeration which you can check for 'address already in use'. If you do something like this; it would be great to have contributed back to suave :)

**Comment by Krzysztof-Cieslak** [comment]
This is partially solved in `1.4.7` - port is chosen randomly from range 8999 - 8100. Not most elegant solution but it should work right now for most cases (in worst case it just requires restart of Atom). In longer terms it should be probably fixed in FSAC or in Sauve (as @haf suggested)
